backend/src/index.spec.ts

```diff
@@ -1,28 +1,52 @@
 import { resolvers } from './resolvers'
 import { typeDefs } from './typeDefs'
+import { permissions } from './permissions'
 import { PostsDataSource } from './postsDataSource'
 import { ApolloServer } from 'apollo-server'
 import { createTestClient } from 'apollo-server-testing'
+import { applyMiddleware } from 'graphql-middleware'
+import { makeExecutableSchema } from 'graphql-tools'
+const bcrypt = require('bcrypt')
+const jwt = require('jsonwebtoken')
+require('dotenv').config()
+
+const testContext = (testToken?) => {
+  let token = testToken || ''
+  token = token.replace('Bearer ', '')
+  try {
+    const decodedJwt = jwt.verify(
+      token,
+      process.env.JWTSECRET
+    )
+    return { decodedJwt }
+  } catch (e) {
+    return {}
+  }
+}
+const setupServer = (postData, userData, testToken?) => {
```

Comment:
There is nothing wrong with calling `setupServer` for every test case, it will guarantee you atomic tests. However, we know that the only state that needs to be re-initialized is `context` and `dataSources`. So you could consider to share the `server` for every test case and re-initialize only those parts that are not stateless.